ZBED4.py

```diff
@@ -0,0 +1,257 @@
+        DNA_ALPHABET (set): DNA nucleotides
+        RNA_ALPHABET (set): RNA nucleotides
+        PROTEIN_ALPHABET (set): proteinogenic aminoacids
+
+        NOTE: DNA_ALPHABET, RNA_ALPHABET, PROTEIN_ALPHABET are PRIVATE variables
+        """
+        self.seq = seq
+        self.type = type
+        self.__DNA_ALPHABET = {"A", "T", "G", "C"}
+        self.__RNA_ALPHABET = {"A", "U", "G", "C"}
+        self.__PROTEIN_ALPHABET = {"A","C","D","E","F","G","H","I","K","L","M",
+                    "N","P","Q","R","S","T","V","W","Y"}
+
+    def __len__(self) -> int:
+        """Count length of correct sequence."""
+        if self.is_type():
+            return len(self.seq)
+    
+    def seq_to_print(self) -> str:
+        """Removes \t, \n and " " in sequence."""
+        return self.seq.replace('\n', '').replace('\t', '').replace(" ", '').upper()
+
+    def get_subseq(self, start:int, stop=None) -> list:
+        """Make slice of sequence or return one letter if stop is None as default.
+        """
+        if self.is_type():
+            if stop is None:
+                return self.seq[start]
+            return self.seq[start:stop:1]
+    
+    def is_type(self) -> bool:
+        """Checks the correctness of the correspondence between type and sequence."""
+        if self.type == 'DNA':
+            return set(self.seq.upper()) <= self.__DNA_ALPHABET
+        elif self.type == 'RNA':
+            return set(self.seq.upper()) <= self.__RNA_ALPHABET
+        elif self.type == 'protein':
+            return set(self.seq.upper()) <= self.__PROTEIN_ALPHABET
+        elif self.type not in ('DNA', 'RNA', 'protein'):
+            raise ValueError(f'Unknown type ({self.type}) of biological sequence!')    
+
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    A child class of "BiologicalSequence" that implements work with nucleic acids.
+
+    Methods:
+    + gc_content(self) : count G and C in sequence and return result in %
+    + complement(self) : find compliment sequence
+
+    NOTE: correct sequence maens that is_type(self) is True
+    """
+    def __init__(self, seq:str, type:str) -> None:
+        """
+        RNA_COMPLEMENT_NUCLS (dict): compliment nucliotides of RNA in upper and lower cases
+        DNA_COMPLEMENT_NUCLS (dict): compliment nucliotides of DNA in upper and lower cases
+        """
+        super().__init__(seq, type)
+        self.__RNA_COMPLEMENT_NUCLS = { "A":"U", "U":"A", "a":"u", "u":"a", "G":"C", "C":"G", "c":"g", "g":"c"}
+        self.__DNA_COMPLEMENT_NUCLS = {"A":"T", "T":"A", "a": "t", "t":"a", "G":"C", "C":"G", "g":"c", "c":"g"}
+
+    def gc_content(self) -> float:
+        """Count G and C in sequence and return result in %."""
+        if self.is_type():
+            gc_amount = 0
+            for nucl in self.seq:
+                if nucl == "G" or nucl == "C":
+                    gc_amount += 1
+            
+            cur_gc_level = gc_amount / self.seq.__len__() * 100
```

Comment:
Можно написать менее время затратно:
```suggestion
gc_count = sum(1 for base in self.seq if base in 'GC')
gc_count / len(self.seq) * 100
```